src/include/ZividPython/Wrappers.h

```diff
@@ -13,6 +16,50 @@
 
 namespace ZividPython
 {
+    namespace
+    {
+        std::string toSnakeCase(const std::string capitalized)
+        {
+            if(capitalized.empty())
+            {
+                throw std::invalid_argument{ "String is empty." };
+            }
+
+            if(!isupper(capitalized[0]))
+            {
+                throw std::runtime_error{ "First character of string: '" + capitalized + "' is not capitalized" };
+            }
+            std::stringstream ss;
+            ss << char(tolower(capitalized[0]));
+
+            for(auto i = 1; i < capitalized.size(); ++i)
+            {
+                if(isupper(capitalized[i]))
+                {
+                    auto previous = i - 1;
+                    auto next = i + 1;
+
+                    // if surrounded by capital case (looking at the B character): ABC -> abc
+                    if(isupper(capitalized[previous]) && (next < capitalized.size() && isupper(capitalized[next]))
```

Comment:
Should we consider using c++ regex as an more advanced approach?

Reply:
I am open for alternatives, but I think this code is easy to understand and modify.

Reply:
Agree. The code is clear.
I was more thinking in general ... if we need more complex handling in future.
Let's not over-engineer :).